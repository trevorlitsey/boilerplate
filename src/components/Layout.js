import React from 'react';
import { object } from 'prop-types';
import styled from 'styled-components';

import Nav from './Nav';

const Wrapper = styled.div`
	max-width: 1600px;
	margin: auto;
	padding: 40px;
`

const Layout = ({ location, children }) => (
	<div>
		<Nav location={location} />
		<Wrapper>
			{children}
		</Wrapper>
	</div>
)

Layout.propTypes = {
	location: object.isRequired,
}

export default Layout