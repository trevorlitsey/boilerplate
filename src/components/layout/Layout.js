import React from 'react';
import { object } from 'prop-types';
import styled from 'styled-components';

import Nav from './Nav';
import Footer from './Footer';

const Wrapper = styled.div`
	max-width: 1600px;
	min-height: 95vh;
	margin: 0 auto;
	padding: 4%;
`

const Layout = ({ location, user, children }) => (
	<div>
		<Nav location={location} user={user} />
		<Wrapper>
			{children}
		</Wrapper>
		<Footer />
	</div>
)

Layout.propTypes = {
	location: object.isRequired,
	user: object,
}

export default Layout