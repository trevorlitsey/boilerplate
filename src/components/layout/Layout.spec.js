import React from 'react';
import { shallow } from 'enzyme';

import Layout from './Layout';

describe('Layout', () => {

	it('should render without crashing', () => {
		const wrapper = renderLayout();
		expect(wrapper).toMatchSnapshot();
	})

})


// -----------
function renderLayout(props = {}) {
	const propsToUser = {
		location: {
			hash: '',
			key: 'jt5a16',
			pathname: '/snippets',
			search: ''
		}
	}

	return shallow(<Layout {...propsToUser} />);
}