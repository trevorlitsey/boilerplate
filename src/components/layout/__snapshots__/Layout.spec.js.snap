// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Layout
    location={
      Object {
        "hash": "",
        "key": "jt5a16",
        "pathname": "/snippets",
        "search": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Nav
          location={
            Object {
              "hash": "",
              "key": "jt5a16",
              "pathname": "/snippets",
              "search": "",
            }
          }
          user={undefined}
        />,
        <styled.div />,
        <Footer />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "location": Object {
            "hash": "",
            "key": "jt5a16",
            "pathname": "/snippets",
            "search": "",
          },
          "user": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Nav
            location={
              Object {
                "hash": "",
                "key": "jt5a16",
                "pathname": "/snippets",
                "search": "",
              }
            }
            user={undefined}
          />,
          <styled.div />,
          <Footer />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "location": Object {
              "hash": "",
              "key": "jt5a16",
              "pathname": "/snippets",
              "search": "",
            },
            "user": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
